src/common/com/intellij/plugins/haxe/model/HaxeAbstractClassModel.java

```diff
@@ -57,24 +57,32 @@ public HaxeClass getUnderlyingClass(@Nullable final HaxeGenericResolver resolver
         }
         List<HaxeGenericParamModel> params = getGenericParams();
         if (!params.isEmpty()) {
-          ResultHolder result = resolver.resolve(params.get(0).getName());
-          SpecificHaxeClassReference ref = null != result ? result.getClassType() : null;
-          HaxeClass underlyingClass = ref != null ? ref.getHaxeClass() : null;
-          return underlyingClass;
+          return resolveGeneric(params.get(0).getName(), resolver);
         }
       }
       return null;
     }
     HaxeTypeOrAnonymous anonymous = underlyingTypePsi.getTypeOrAnonymous();
     final HaxeType underlyingType = anonymous != null ? anonymous.getType() : null;
     if (underlyingType != null) {
-      final HaxeClassResolveResult result = underlyingType.getReferenceExpression().resolveHaxeClass();
-      return result.getHaxeClass();
+      HaxeReferenceExpression referenceExpression = underlyingType.getReferenceExpression();
+      final HaxeClassResolveResult result = referenceExpression.resolveHaxeClass();
+      HaxeClass resultClass = result.getHaxeClass();
+      return null != resultClass ? resultClass : resolveGeneric(referenceExpression.getIdentifier().getText(), resolver);
```

Comment:
Simple fix.  If we weren't able to resolve the underlying class, check against the generic specialization/resolver.